// Copyright (c) 2012 The Chromium Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

syntax = "proto2";

option optimize_for = LITE_RUNTIME;

package enterprise_management;

message DevicePolicyRefreshRateProto {
  // In milliseconds.
  optional int64 device_policy_refresh_rate = 1;
}

message UserWhitelistProto {
  repeated string user_whitelist = 1;
}

message AllowNewUsersProto {
  // Determines whether we allow arbitrary users to log into the device.
  // Should default to true.
  // This interacts with the UserWhitelistProto as follows:
  // allow_new_users | user_whitelist | user_whitelist_size | anyone can log in
  //-----------------+----------------+---------------------+------------------
  //  present, true  | not present    | N/A                 | Yes
  //-----------------+----------------+---------------------+------------------
  //  present, true  | present        | >= 0                | Yes
  //-----------------+----------------+---------------------+------------------
  //  present, false | not present    | N/A                 | (Broken) Yes
  //-----------------+----------------+---------------------+------------------
  //  present, false | present        | 0                   | (Broken) Yes
  //-----------------+----------------+---------------------+------------------
  //  present, false | present        | > 0                 | No, W/L enforced
  //-----------------+----------------+---------------------+------------------
  //  not present    | not present    | N/A                 | Yes
  //-----------------+----------------+---------------------+------------------
  //  not present    | present        | 0                   | Yes
  //-----------------+----------------+---------------------+------------------
  //  not present    | present        | > 0                 | No, W/L enforced
  //-----------------+----------------+---------------------+------------------
  optional bool allow_new_users = 1;
}

message GuestModeEnabledProto {
  // Determines if guests are allowed to log in to the device.
  // Should default to true.
  optional bool guest_mode_enabled = 1;
}

message ShowUserNamesOnSigninProto {
  // Determines if we show pods for existing users on the sign in screen.
  // Should default to true.
  optional bool show_user_names = 1;
}

message DataRoamingEnabledProto {
  // Determines if cellular data roaming is enabled.  Should default to false.
  optional bool data_roaming_enabled = 1;
}

message DeviceProxySettingsProto {
  // One of "direct", "auto_detect", "pac_script", "fixed_servers", "system"
  optional string proxy_mode = 1;
  optional string proxy_server = 2;
  optional string proxy_pac_url = 3;
  optional string proxy_bypass_list = 4;
}

message CameraEnabledProto {
  optional bool camera_enabled = 1;
}

message MetricsEnabledProto {
  optional bool metrics_enabled = 1;
}

message ReleaseChannelProto {
  // One of "stable-channel", "beta-channel", or "dev-channel"
  optional string release_channel = 1;
}

message DeviceOpenNetworkConfigurationProto {
  // The network configuration blob. This is a JSON string as specified by ONC.
  optional string open_network_configuration = 1;
}

// Policies to turn on portions of the device status reports.
message DeviceReportingProto {
  optional bool report_version_info = 1;
  optional bool report_activity_times = 2;
  optional bool report_boot_mode = 3;
}

message ChromeDeviceSettingsProto {
  optional DevicePolicyRefreshRateProto device_policy_refresh_rate = 1;
  optional UserWhitelistProto user_whitelist = 2;
  optional GuestModeEnabledProto guest_mode_enabled = 3;
  optional DeviceProxySettingsProto device_proxy_settings = 4;
  optional CameraEnabledProto camera_enabled = 5;
  optional ShowUserNamesOnSigninProto show_user_names = 6;
  optional DataRoamingEnabledProto data_roaming_enabled = 7;
  optional AllowNewUsersProto allow_new_users = 8;
  optional MetricsEnabledProto metrics_enabled = 9;
  optional ReleaseChannelProto release_channel = 10;
  optional DeviceOpenNetworkConfigurationProto open_network_configuration = 11;
  optional DeviceReportingProto device_reporting = 12;
}
